flaml/model.py

```diff
@@ -2099,6 +2099,129 @@ class XGBoostLimitDepth_TS(TS_SKLearn):
+        https://auto.gluon.ai/stable/tutorials/text_prediction/customization.html#custom-hyperparameter-values
+        """
+        # TODO: expand the search space
+        search_space_dict = {
+            "model.network.agg_net.mid_units": {
+                "domain": tune.choice(list(range(32, 129))),
+                "init_value": 128,
+            },
+            "optimization.lr": {
+                "domain": tune.loguniform(lower=1E-5, upper=1E-4),
+                "init_value": 1E-4,
+            },
+            "optimization.wd": {
+                "domain": tune.choice([1E-4, 1E-3, 1E-2]),
+                "init_value": 1E-4,
+            },
+            "optimization.warmup_portion": {
+                "domain": tune.choice([0.1, 0.2]),
+                "init_value": 0.1, 
+            },
+            "optimization.layerwise_lr_decay": {
+                "domain": tune.choice([0.8, 0.9]),
+                "init_value": 0.8,
+            },
+            "optimization.nbest": {
+                "domain": tune.choice([2, 3, 4,]),
+                "init_value": 3,
+            },
+            "optimization.num_train_epochs": {
+                "domain": tune.choice([5, 10, 15,]),
+                "init_value": 10,
+            },
+            "optimization.per_device_batch_size": {
+                "domain": tune.choice([2, 4, 8,]),
+                "init_value": 10,
+            },
+            "optimization.batch_size": {
+                "domain": tune.choice([32, 64, 128,]),
+                "init_value": 128,
+            },
+        }
+        return search_space_dict
+
+    def _init_ag_args(self, automl_fit_kwargs: dict = None):
+        from .nlp.utils import AGArgs
+
+        ag_args = AGArgs()
+        for key, val in automl_fit_kwargs["ag_args"].items():
+            assert (
+                key in ag_args.__dict__
+            ), "The specified key {} is not in the argument list of flaml.nlp.utils::AGArgs".format(
+                key
+            )
+            setattr(ag_args, key, val)
+        self.ag_args = ag_args
+
+    def _set_seed(self, seed):
+        import random
+        import mxnet as mx
+        # NOTE: if support pytorch backend, uncomment below
+        # import torch as th
+        # th.manual_seed(seed)
+        mx.random.seed(seed)
+        np.random.seed(seed)
+        random.seed(seed)
+
+    def fit(self, X_train=None, y_train=None, budget=None, **kwargs):
+        from autogluon.text import TextPredictor
+
+        self._kwargs = kwargs
+        self._init_ag_args(kwargs)
+        seed = self._kwargs.get("seed", 123)
+        self._set_seed(seed)
+
+        assert (self.ag_args.backend == "mxnet"), "the pytorch automm model is not supported. "
+        # get & set the hyperparameters, update with self.params
+        hyperparameters = self.ag_args.get_presets()
+        search_space = hyperparameters["models"]["MultimodalTextModel"]["search_space"]
+        for key, value in self.params.items():

```

Comment:
Instead of a for loop, try implement using one line with dict.update()

Reply:
Got it, will fix.

Reply:
There are two problems here: 
1. flaml uses np.float64 but AG uses float, need to convert or else will raise error; 
3. there will be a "n_jobs" in self.params, which should not be passed to "search_space" of "hyperparameters".

Reply:
both problems can be solved using python conditional value assignment https://stackoverflow.com/questions/6402311/python-conditional-assignment-operator